### ISSUE TITLE
Resources should show something more descriptive

### ISSUE THREAD

**Issue by kennyw12** [concourse/concourse#310]
Currently, when a user wants to see which commit a git resource is using, the first thing we see is the hash and that doesn't really tell us anything. It would be nice if it displayed the commit message or part of it. 

![screen shot 2016-03-09 at 11 44 55 am](https://cloud.githubusercontent.com/assets/1689040/13648221/6dd464b8-e5ec-11e5-9c69-19fdef33ae40.png)

**Comment by vito** [comment]
Do you mean where the version is? That feels like it'd get pretty noisy which won't help when you're looking for a particular sha. The metadata includes the message. I'd rather change the UI (perhaps have everything expanded by default) than change the resource.

**Comment by ansel1** [comment]
What if it showed the nearest tag/branch? There's a git log option for
that.

**Comment by vito** [comment]
Perhaps, but the branch should already be known at least (it's part of the resource). Again I'd try to reframe this as a UI issue that we can fix instead of changing a resource for cosmetic reasons.

**Comment by kennyw12** [comment]
Yes, I agree that this is a UI issue. I was referring to the `ref <commit-hash>'. The UI issue is when we push to a branch we aren't sure if concourse picked up the updated resource until we click into the ref and refresh the page a bunch.

**Comment by donaldguy** [comment]
I'm by no means inclined to work on it now, but presuming there is a way to pass args to check(?, I'm new here), you would just have to optionally call out to [`git describe`](https://git-scm.com/docs/git-describe).

**Comment by DenisIzmaylov** [comment]
+1

**Comment by vito** [comment]
Generalized this a bit; I think it'd be nice if resources could emit semantic metadata that we could then show in a nicer way on the resource history page. For example, commit messages, icons (e.g. GitHub gravatars), clickable URLs, etc.

For the `git` resource, this could look something like:

```json
{
  "version": {
    "ref": "eac641a8b474570ff45594f14c379d8d6fce0233"
  },
  "metadata": [
    {
      "icon": "https://avatars1.githubusercontent.com/u/1880?v=3&s=96",
      "url": "https://github.com/vito",
      "title": "vito"
    },
    {
      "title": "100x larger events batch size",
      "url": "https://github.com/concourse/atc/commit/eac641a8b474570ff45594f14c379d8d6fce0233"
    }
  ]
}
```

Then, each of these metadata fields would be rendered columnwise in the UI, like:

```
| [icon] vito | 100x larger events batch size | ref: eac641a8b474570ff45594f14c379d8d6fce0233 |
```

Obviously we'd want to spend a bit more time on the schema, but that's what I'm thinkin' of, anyway.

**Comment by vito** [comment]
Related to https://github.com/concourse/concourse/issues/534, which also proposes that we collect metadata as part of `check`.

**Comment by jchesterpivotal** [comment]
Possibly related to https://github.com/concourse/concourse/issues/2900

**Comment by vito** [comment]
Not really 'planned' just yet - we took semantic metadata out of the first v2 resources proposal - but I'll put it in the Resources v2 project to keep the thought alive.

**Comment by speculatrix** [comment]
this could be tied into #569  https://github.com/concourse/concourse/issues/569

**Comment by taylorsilva** [comment]
# Issue Bankruptcy

This issue is being closed as part of 2025's **Issue Bankruptcy** declaration. [See here for details](https://blog.concourse-ci.org/posts/2025-01-06-concourse-2024-review/)

If you think this issue still has merit then feel free to open a new issue and reference this one.
